sys/mips/mips/pmap.c

```diff
@@ -2918,6 +2969,34 @@ pmap_testbit(vm_page_t m, int bit)
 	return (rv);
 }
 
+/*
+ * Like pmap_testbit, but inverted (that is, hunts for a 0 bit, rather than
+ * a 1).
+ */
+static boolean_t
+pmap_testbit_inv(vm_page_t m, pt_entry_t bit)
+{
+	pv_entry_t pv;
+	pmap_t pmap;
+	pt_entry_t *pte;
+	int rv = 1;
+
+	if (m->oflags & VPO_UNMANAGED)
+		return (rv == 1);
+
+	rw_assert(&pvh_global_lock, RA_WLOCKED);
+	TAILQ_FOREACH(pv, &m->md.pv_list, pv_list) {
+		pmap = PV_PMAP(pv);
+		PMAP_LOCK(pmap);
+		pte = pmap_pte(pmap, pv->pv_va);
+		rv = pte_test(pte, bit);
+		PMAP_UNLOCK(pmap);
+		if (rv == 0)
+			break;
+	}
+	return (rv == 1);
```

Comment:
This function will currently return true if there are no PTEs, but if there are PTEs it will return true if "bit" is set for all of them and false otherwise. I think that is not the intended behavior.


```suggestion
	int rv = 0;

	if (m->oflags & VPO_UNMANAGED)
		return TRUE;

	rw_assert(&pvh_global_lock, RA_WLOCKED);
	TAILQ_FOREACH(pv, &m->md.pv_list, pv_list) {
		pmap = PV_PMAP(pv);
		PMAP_LOCK(pmap);
		pte = pmap_pte(pmap, pv->pv_va);
		rv = pte_test(pte, bit);
		PMAP_UNLOCK(pmap);
		if (rv == 0)
			break;
	}
	return (rv == 0);
```

Reply:
Good catch; thank you.  This turns out to no longer be relevant for this PR, but I've fixed the follow-on commit that'll come with revocation.